tests/test_tools.py

```
@@ -0,0 +1,310 @@
+
+            @session.tool(description="Calculate math expression")
+            def calculate(expression: str) -> str:
+                results['called'] = True
+                # Simple calculator (in real code, use safe evaluation)
+                if "2 + 2" in expression:
+                    return "4"
+                return "calculated"
+
+            response = session.generate("What is 2 + 2?")
+
+            assert results.get('called') == True
```

Ruff (0.14.3)

Avoid equality comparisons to `True`; use `results.get('called'):` for truth checks

Replace with `results.get('called')`

(E712)